src/NuGet.Core/NuGet.Packaging/Signing/Utility/CertificateChainUtility.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    public static class CertificateChainUtility
+    {
+        /// <summary>
+        /// Create a list of certificates in chain order with the leaf first and root last.
+        /// </summary>
+        public static IReadOnlyList<X509Certificate2> GetCertificateChainForSigning(
+            X509Certificate2 certificate,
+            X509Certificate2Collection extraStore,
+            NuGetVerificationCertificateType certificateType)
+        {
+            if (certificate == null)
+            {
+                throw new ArgumentNullException(nameof(certificate));
+            }
+
+            if (extraStore == null)
+            {
+                throw new ArgumentNullException(nameof(extraStore));
+            }
+
+            using (var chain = new X509Chain())
+            {
+                SetCertBuildChainPolicy(
+                    chain.ChainPolicy,
+                    extraStore,
+                    DateTime.Now,
+                    certificateType);
+
+                if (BuildCertificateChain(chain, certificate, out var chainStatuses))
+                {
+                    return CertificateChainToList(chain);
+                }
+
+                var messages = GetMessagesFromChainStatuses(chainStatuses);
+
+                throw new SignatureException(NuGetLogCode.NU3018, string.Format(CultureInfo.CurrentCulture, Strings.ErrorInvalidCertificateChain, string.Join(", ", messages)));
+            }
+        }
+
+
+        /// <summary>
+        /// Create an ordered list of certificates. The leaf node is returned first.
+        /// </summary>
+        /// <remarks>This does not check validity or trust. It returns the chain as-is.</remarks>
+        public static IReadOnlyList<X509Certificate2> CertificateChainToList(X509Chain certChain)
```

Comment:
The method name should start with a verb.  The old `GetCertificateChain(...)` is a good model.  Perhaps `GetCertificateChainList(...)`?